src/test/scala/org/ergoplatform/api/routes/ScriptApiRouteSpec.scala

```diff
@@ -42,6 +42,22 @@ class ScriptApiRouteSpec  extends FlatSpec
       |}
       |""".stripMargin
 
+  it should "execute script with context" in {
+    val suffix = "/executeWithContext"
+    val stream = ClassLoader.getSystemClassLoader.getResourceAsStream("execute-script.json")
+    val req = scala.io.Source.fromInputStream(stream).getLines().mkString("\n")
+    val assertion = (json: Json) => {
+      status shouldBe StatusCodes.OK
+      val value = json.hcursor.downField("value").downField("op").as[Int].right.get
+      val condition = json.hcursor.downField("value").downField("condition").as[Boolean].right.get
+      val cost = json.hcursor.downField("cost").as[Int].right.get
+      value shouldEqual -45
+      condition shouldEqual true
+      cost shouldEqual 40
+    }
+    val json = io.circe.parser.parse(req)
+    Post(prefix + suffix, json) ~> route ~> check(assertion(responseAs[Json]))
+  }
 
```

Comment:
In addition to the checks above, it make sense to call reduceToCrypto directly (i.e. without API) and compare the results.

Reply:
What kind of functionality we will test in this case? Correct transfer of JSON response though HTTP protocol? Why don't we do in other similar requests (like p2sAddress)?

Reply:
Yes, it is what API for. 
It opens access to internal components, and we need a collection of scripts which checks that
executing via API is equivalent to executing directly for every script in the collection.
Here we assume that executing directly is ground truth, since the Interpreter is tested in sigma.
To do this, you actually need to be able to create different contexts.